modules/platform/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/bulk/AutomationBulkActionUi.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+package org.nuxeo.ecm.automation.core.operations.services.bulk;
+
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+
+/**
+ * Bulk Action that runs an automation operation dedicated for UI
+ *
+ * @since 11.5
+ */
+public class AutomationBulkActionUi extends AutomationBulkAction {
+    public static final String ACTION_NAME = "automationUi";
+    private static final String QUERY_LIMIT_PREFIX = "org.nuxeo.ecm.automation.ui.bulk.queryLimit.";
+
+    public static int getQueryLimit(String operationId) {
```

Comment:
Shouldn't this be in the abstract class? We can impose a limit to the regular `automation` action as well via param through the REST API, right?

Reply:
Only the `automation-ui` action is using a query limit specified by the configuration service. 
Yes for the `automation` action you can use the `queryLimit` parameters.